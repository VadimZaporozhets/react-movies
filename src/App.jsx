import React from 'react';
import { GeneralComponent, FunctionalComponent, Pure } from './components';

export const App = () => (
    <>
        <GeneralComponent />
        <Pure />
        <FunctionalComponent />
    </>
);
