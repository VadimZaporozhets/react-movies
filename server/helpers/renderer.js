import React from 'react';
import { renderToString } from 'react-dom/server';
import { StaticRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import { renderRoutes } from 'react-router-config';

import { routes } from '../../src/routes';
import { getInitialTemplate } from '../page-template';

export const renderer = (req, store, context) => {
    const content = renderToString(
        <Provider store={store}>
            <StaticRouter location={req.path} context={context}>
                {renderRoutes(routes)}
            </StaticRouter>
        </Provider>
    );

    return getInitialTemplate(content, store);
};
