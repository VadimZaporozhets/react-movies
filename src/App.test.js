import React from 'react';
import { App } from './App';
import renderer from 'react-test-renderer';

describe('App component', () => {
    it('should render app container', () => {
        const wrapper = renderer.create(<App />).toJSON();

        expect(wrapper).toMatchSnapshot();
    });
});