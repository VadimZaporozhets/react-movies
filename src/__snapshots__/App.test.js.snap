// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component should render app container 1`] = `
Array [
  <h2>
    Hello from Component
  </h2>,
  <h2>
    Hello from Pure Component
  </h2>,
  <h2>
    Hello from functional Component
  </h2>,
]
`;
